src/aat/java/uk/gov/hmcts/reform/em/annotation/functional/RectangleScenarios.java

```diff
@@ -0,0 +1,387 @@
+        final String newAnnotationSetId = createAnnotationSet();
+        final String annotationId = createAnnotation(newAnnotationSetId);
+        final String rectangleId = UUID.randomUUID().toString();
+        final JSONObject rectangle = createRectanglePayload(annotationId, rectangleId);
+
+        rectangle.remove("id");
+
+        request
+                .body(rectangle.toString())
+                .put("/api/rectangles")
+                .then()
+                .statusCode(400)
+                .log().all();
+    }
+
+    @Test
+    public void shouldReturn401WhenUpdateRectangle() {
+        final String annotationId = UUID.randomUUID().toString();
+        final String rectangleId = UUID.randomUUID().toString();
+        final JSONObject rectangle = createRectanglePayload(annotationId, rectangleId);
+
+        unAuthenticatedRequest
+                .body(rectangle.toString())
+                .put("/api/rectangles")
+                .then()
+                .statusCode(401)
+                .log().all();
+    }
+
+    @Test
+    public void shouldReturn200WhenDeleteRectangleById() {
+        final String newAnnotationSetId = createAnnotationSet();
+        final String annotationId = createAnnotation(newAnnotationSetId);
+        final String rectangleId = UUID.randomUUID().toString();
+        final ValidatableResponse createdResponse = createRectangle(annotationId, rectangleId);
+        final String id = extractJSONObjectFromResponse(createdResponse).getString("id");
+
+        final ValidatableResponse deletedResponse = deleteRectangleById(id);
+
+        deletedResponse.statusCode(200);
+    }
+
+    @Test
+    public void shouldReturn500WhenDeleteRectangleById() {
+        final String rectangleId = UUID.randomUUID().toString();
+        final ValidatableResponse deletedResponse = deleteRectangleById(rectangleId);
+
+        deletedResponse.statusCode(500);
+    }
+
+    @Test
+    public void shouldReturn401WhenDeleteRectangle() {
+        unAuthenticatedRequest
+                .delete("/api/rectangles/" + UUID.randomUUID())
+                .then()
+                .statusCode(401)
+                .log().all();
+    }
+
+    @Test
+    public void shouldReturn200WhenTryToUpdateRectangleAfterItHasBeenDeleted() {
+        final String newAnnotationSetId = createAnnotationSet();
+        final String annotationId = createAnnotation(newAnnotationSetId);
+        final String rectangleId = UUID.randomUUID().toString();
+        final ValidatableResponse response = createRectangle(annotationId, rectangleId);
+        final JSONObject rectangle = extractJSONObjectFromResponse(response);
+        final String id = rectangle.getString("id");
+        deleteRectangleById(id).statusCode(200);
+        rectangle.put("x", 3f);
+        rectangle.put("y", 4f);
+
+        request
+                .body(rectangle.toString())
+                .put("/api/rectangles")
+                .then()
+                .statusCode(200)
```

Comment:
shouldn't this return 404 since the resource doesn't exist?

Reply:
At the moment update is creating new resource when it does not find the entity with given id and updating if it found.